React18 support

**Before you submit:**

- [:heavy_check_mark: ] Please search through the existing issues (both open AND closed) to see if your issue has been discussed before. GitHub issue search can be used for this: https://github.com/imgix/react-imgix/issues?utf8=%E2%9C%93&q=is%3Aissue
- [ :heavy_check_mark:] Please ensure the problem has been isolated and reduced. This link explains more: http://css-tricks.com/6263-reduced-test-cases/

**Describe the bug**
When using React@18 I get the following error while installing react-imgix:

```
...
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR!
npm ERR! peer react@"15.x || 16.x || 17.x" from react-imgix@9.5.0
npm ERR! node_modules/react-imgix
npm ERR!   react-imgix@"^9.5.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
...
```

**To Reproduce**
Install latest React. Install react-imgix.

**Expected behaviour**
Should install react-imgix without any problems.

**Screenshots**
N/A

**Information:**

- react-imgix version: v9.5

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  node: circleci/node@5.0.0
+  node: circleci/node@5.0.2
   browser-tools: circleci/browser-tools@1.2.4
 
 jobs:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -25,7 +25,7 @@ jobs:
           name: Check install
       - when:
           condition:
-            equal: [ "current", <<parameters.version>>]
+            equal: [ "17.9", <<parameters.version>>]
           steps:
             # Prevents build error on stable node version
             - run: echo 'export NODE_OPTIONS=--openssl-legacy-provider' >> $BASH_ENV
@@ -33,7 +33,7 @@ jobs:
       - run: npm run test
   deploy:
     docker:
-      - image: cimg/node:current
+      - image: cimg/node:17.9
     steps:
       - checkout
       - node/install-packages
@@ -46,7 +46,7 @@ workflows:
           matrix:
             parameters:
               version:
-                - "current"
+                - "17.9"
                 - "lts"
       - deploy:
           requires:

--- a/package.json
+++ b/package.json
@@ -51,8 +51,8 @@
     "test": "jest && npm run test:browser:all"
   },
   "peerDependencies": {
-    "react": "15.x || 16.x || 17.x",
-    "react-dom": "15.x || 16.x || 17.x"
+    "react": "15.x || 16.x || 17.x || 18.x",
+    "react-dom": "15.x || 16.x || 17.x || 18.x"
   },
   "dependencies": {
     "@imgix/js-core": "^3.1.3",

